src/main/kotlin/com/autonomousapps/internal/analyzer/JvmProjectAnalyzer.kt

```diff
@@ -104,7 +104,7 @@ internal abstract class JvmAnalyzer(
   }
 }
 
-internal class JavaAppAnalyzer(
+internal class JavaWithoutAbiAnalyzer(
```

Comment:
can application projects reasonably have an `api`?

Reply:
No. But source sets inside a library project (like integrationTest) also do not have an API. That's why I changed the name as it is used for **all** source sets without API now.

Reply:
Well, maybe an application can have an `api` if the user decides so. E.g. you have something that can run, but you still want to use it somewhere else as a library. You could configure it like this.

I wonder if we really still need to distinguish between `application` and `library` cases or if checking it the `...Api` config exists is enough to make a distinction _per source set_.

Reply:
One last comment 😄 

This PR adds the `val hasAbi = configurations.findByName(sourceSet.apiConfigurationName) != null` check to the "application" case as it is now (only not in the SPRING_BOOT_PLUGIN special case).

I think it is correct to do so, as I wrote in the second comment.